packages/x-charts/src/BarChart/BatchBarPlot/BatchBarPlot.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+  selectorChartIsSeriesFaded,
+  selectorChartIsSeriesHighlighted,
+  selectorChartSeriesHighlightedItem,
+  selectorChartSeriesUnfadedItem,
+  type UseChartHighlightSignature,
+} from '../../internals/plugins/featurePlugins/useChartHighlight';
+import { useStore } from '../../internals/store/useStore';
+import { useOnItemClick } from '../useOnItemClick';
+import { useInteractionItemProps } from '../useItemInteractionProps';
+import { createPath, useCreateBarPaths } from './useCreateBarPaths';
+import { BarGroup } from './BarGroup';
+
+interface BatchBarPlotProps extends IndividualBarPlotProps {}
+
+export function BatchBarPlot({
+  completedData,
+  borderRadius = 0,
+  onItemClick,
+  skipAnimation = false,
+}: BatchBarPlotProps) {
+  const { onClick, onPointerUp } = useOnItemClick(onItemClick);
+  const interactionItemProps = useInteractionItemProps(onItemClick != null);
+
+  return (
+    <React.Fragment>
+      {completedData.map((series) => (
+        <SeriesBatchPlot
+          key={series.seriesId}
+          series={series}
+          borderRadius={borderRadius}
+          skipAnimation={skipAnimation}
+        />
+      ))}
+      <DrawingAreaRect onClick={onClick} onPointerUp={onPointerUp} {...interactionItemProps} />
+    </React.Fragment>
+  );
+}
+
+const MemoFadedHighlightedBars = React.memo(FadedHighlightedBars);
+
+function SeriesBatchPlot({
+  series,
+  borderRadius,
+  skipAnimation,
+}: {
+  series: ProcessedBarSeriesData;
+  borderRadius: number;
+  skipAnimation: boolean;
+}) {
+  const classes = useUtilityClasses();
+  const { store } = useChartContext<[UseChartHighlightSignature]>();
+  const isSeriesHighlighted = useSelector(store, selectorChartIsSeriesHighlighted, series.seriesId);
+  const isSeriesFaded = useSelector(store, selectorChartIsSeriesFaded, series.seriesId);
+
+  return (
+    <React.Fragment>
+      <BarGroup
+        className={classes.series}
+        data-series={series.seriesId}
+        layout={series.layout}
+        xOrigin={series.xOrigin}
+        yOrigin={series.yOrigin}
+        skipAnimation={skipAnimation}
+        data-faded={isSeriesFaded || undefined}
+        data-highlighted={isSeriesHighlighted || undefined}
+      >
+        <BatchBarSeriesPlot processedSeries={series} borderRadius={borderRadius} />
+      </BarGroup>
+      <MemoFadedHighlightedBars processedSeries={series} borderRadius={borderRadius} />
+    </React.Fragment>
+  );
+}
+
+/**
+ * A transparent rectangle that covers the drawing area of the chart to capture clicks.
+ */
+function DrawingAreaRect(props: React.HTMLAttributes<SVGRectElement>) {
+  const store = useStore();
+  const drawingArea = useSelector(store, selectorChartDrawingArea);
+
+  return (
+    <rect
+      x={drawingArea.left}
+      y={drawingArea.top}
+      width={drawingArea.width}
+      height={drawingArea.height}
+      fill="transparent"
+      {...props}
+    />
+  );
+}
```

Comment:
Why not using the SVG directly to capture click?

If we keep it, this component could be shared with other charts